File: app/src/main/java/com/zl/weilu/androidut/mvp/ui/LoginMVPActivity.java
package com.zl.weilu.androidut.mvp.ui;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.zl.weilu.androidut.R;
import com.zl.weilu.androidut.mvp.base.BaseMVPActivity;

public class LoginMVPActivity extends BaseMVPActivity<LoginMvpView, LoginPresenter> implements LoginMvpView, View.OnClickListener{

    private TextView mTvSendIdentify;
    private EditText mEtMobile;
    private EditText mEtIdentify;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        mEtMobile = this.findViewById(R.id.et_mobile);
        mEtIdentify = this.findViewById(R.id.et_identify);
        mTvSendIdentify = this.findViewById(R.id.tv_send_identify);

        this.findViewById(R.id.tv_login).setOnClickListener(this);
        mTvSendIdentify.setOnClickListener(this);
    }

    @Override
    protected LoginPresenter createPresenter() {
        return new LoginPresenter();
    }

    @Override
    public void countdownComplete() {
        mTvSendIdentify.setText(R.string.login_send_identify);
        mTvSendIdentify.setEnabled(true);
    }

    @Override
    public void countdownNext(String time) {
        mTvSendIdentify.setText(TextUtils.concat(time, "秒后重试"));
    }

    @Override
    public void loginSuccess() {
        
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.tv_send_identify:
                mTvSendIdentify.setEnabled(false);
                mPresenter.getIdentify();
                break;
            case R.id.tv_login:
                mPresenter.login(mEtMobile.getText().toString().trim(),
                        mEtIdentify.getText().toString().trim());
                break;
            default:
                break;
        }
    }
}

File: app/src/main/java/com/zl/weilu/androidut/mvp/ui/LoginMVPActivity.kt
package com.zl.weilu.androidut.mvp.ui

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import com.zl.weilu.androidut.R
import com.zl.weilu.androidut.mvp.base.BaseMVPActivity
import kotlinx.android.synthetic.main.activity_login.*

class LoginMVPActivity : BaseMVPActivity<LoginMvpView, LoginPresenter>(), LoginMvpView, View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        tv_login.setOnClickListener(this)
        tv_send_identify.setOnClickListener(this)
    }

    override fun createPresenter(): LoginPresenter {
        return LoginPresenter()
    }

    override fun countdownComplete() {
        tv_send_identify.setText(R.string.login_send_identify)
        tv_send_identify.isEnabled = true
    }

    override fun countdownNext(time: String) {
        tv_send_identify.text = TextUtils.concat(time, "秒后重试")
    }

    override fun loginSuccess() {
        showToast("登录成功")
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.tv_send_identify -> {
                tv_send_identify.isEnabled = false
                mPresenter!!.getIdentify()
            }
            R.id.tv_login -> mPresenter!!.login(et_mobile.text.toString().trim { it <= ' ' },
                    et_identify.text.toString().trim { it <= ' ' })
            else -> {
            }
        }
    }
}
